### ISSUE TITLE
Cold start optimization: optimistic pod selection

### ISSUE THREAD

**Issue by soamvasani** [fission/fission#204]
Today we relabel a pod synchronously when removing it from the pool.  We can make this async and handle conflicts separately. This will improve the median cold-start time (by a few tens of msec).

**Comment by ajbouh** [comment]
Is the primary risk that we try to specialize a pod that's already been specialized?

If so, it seems like environments should do their own locking in /specialize and return an error if they're specialized more than once. Then the specialization logic in pool should retry. Would you agree with that?

**Comment by soamvasani** [comment]
Yup -- the environments already do that :)  We just need the change to actually retry, and switch to asynchronously relabeling pods in poolmgr.

**Comment by sanketsudake** [comment]
This has been inactive for 3098 days. We're scrubbing the backlog and closing stale items to keep it actionable. If this is still relevant, please reopen or comment and we'll pick it back up.
